16-07-2022/Soal2.java

```diff
@@ -0,0 +1,34 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.StringTokenizer;
+
+public class Soal2 {
+    public static void main(String[] args) {
+        ArrayList<String> token = new ArrayList<String>();
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+        sc.close();
+        StringTokenizer st1 = new StringTokenizer(input);
+        int operasi = 0;
+        for (int i = 0; st1.hasMoreTokens(); i++) {
```

Comment:
dan, kalo diliat disini, variable `i` nyo dak kepake. `hasMoreTokens` itu return boolean, jadi biso dipake cak ini be
`while(st1.hasMoreTokens())`

agek di real case nyo, penting untuk mikirke efisiensi dari code kito. salah satu cara adalah menghindari bikin variable yg dak kepake.

Reply:
selain itu, penting jg untuk tau hasil return dari fungsi-fungsi library yg kito pake. dengan cara liat ke dokumentasi.
contoh:
[https://docs.oracle.com/javase/7/docs/api/java/util/StringTokenizer.html](https://docs.oracle.com/javase/7/docs/api/java/util/StringTokenizer.html)